libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -129,13 +129,34 @@ abstract class DefaultApolloExtension(
     }
 
     project.afterEvaluate {
-      if (registerDefaultService) {
+      @Suppress("DEPRECATION")
+      val hasApolloBlock = !defaultService.graphqlSourceDirectorySet.isEmpty
+          || defaultService.schemaFile.isPresent
+          || !defaultService.schemaFiles.isEmpty
+          || defaultService.alwaysGenerateTypesMatching.isPresent
+          || defaultService.scalarTypeMapping.isNotEmpty()
+          || defaultService.scalarAdapterMapping.isNotEmpty()
+          || defaultService.customScalarsMapping.isPresent
+          || defaultService.customTypeMapping.isPresent
+          || defaultService.excludes.isPresent
+          || defaultService.includes.isPresent
+          || defaultService.failOnWarnings.isPresent
+          || defaultService.generateApolloMetadata.isPresent
+          || defaultService.generateAsInternal.isPresent
+          || defaultService.codegenModels.isPresent
+          || defaultService.addTypename.isPresent
+          || defaultService.generateFragmentImplementations.isPresent
+          || defaultService.requiresOptInAnnotation.isPresent
+          || defaultService.packageName.isPresent
+          || defaultService.packageNameGenerator.isPresent
+
+      if (hasApolloBlock) {
         val packageNameLine = if (defaultService.packageName.isPresent) {
           "packageName.set(\"${defaultService.packageName.get()}\")"
         } else {
           "packageNamesFromFilePaths()"
         }
-        it.logger.warn("""
+        error("""
             Apollo: using the default service is deprecated and will be removed in a future version. Please define your service explicitly:
```

Comment:
I guess the message should be adjusted if it errors:
```suggestion
            Apollo: using the default service is no longer allowed. Please define your service explicitly:
```

Reply:
Right! I went with

```
Apollo: using the default service isdeprecated. Please define your service explicitly:
```